src/parser/DASHTree.cpp

```diff
@@ -1775,19 +1775,54 @@ void DASHTree::RefreshLiveSegments()
 
       for (size_t index{0}; index < updateTree->periods_.size(); index++)
       {
-        if (index == periods_.size()) // Exceeded periods
-          break;
         auto updPeriod = updateTree->periods_[index];
         if (!updPeriod)
           continue;
 
+        Period *period{nullptr};
+        // find matching period based on ID
+        auto itPd = std::find_if(periods_.begin(), periods_.end(),
+                                 [&updPeriod](const Period* item)
+                                 {
+                                   return !item->id_.empty() && item->id_ == updPeriod->id_;
+                                 });
+        if (itPd == periods_.end())
+        {
+          // not found, try matching period based on start
+          itPd = std::find_if(periods_.begin(), periods_.end(),
+                              [&updPeriod](const Period* item)
+                              {
+                                return item->start_ && item->start_ == updPeriod->start_;
+                              });
+
+        }
+        // found!
+        if (itPd != periods_.end())
+          period = *itPd;
+        if (!period && updPeriod->id_.empty() && updPeriod->start_ == 0)
+        {
+          // not found, fallback match based on position
+          if (index < periods_.size())
+            period = periods_[index];
+        }
+        // new period, insert it
+        if (!period)
+        {
+          LOG::LogF(LOGDEBUG, "Inserting new Period (id=%s, start=%ld)", updPeriod->id_.c_str(), updPeriod->start_);
+          std::vector<Period*> tmp = periods_;
+          updateTree->periods_[index] = nullptr;
```

Comment:
i fear that this will cause a memory leak
periods has been stored with `new` operator but you are replacing the pointer without deleting the object
usually should be done
```
delete vector[index]
current iterator = vector.erase(index)
```

Reply:
This is done on purpose, I don't want to delete it I want to take ownership and insert it into the existing period list.